### PR TITLE
#166788774 Endpoint to create a menu template

### PR DESCRIPTION
**Pull request by kimbugp**
#### What does this PR do?
- Add endpoint to create a menu template

#### Description of Task to be completed?
- create menu template endpoints
- add tests for implementations

#### How should this be manually tested?
- Pull and checkout to this branch locally by running `git fetch origin ft-menu-template-endpoint-166788774 && git checkout ft-menu-template-endpoint-166788774`

- Run the server and go to the URL `http://localhost:5000/api/v1/menu_template` to create a menu template 
```
{
  "templateName": "Name of the template",
  "description": "something about the template",
"mealPeriod":"lunch"
}
```
- Run test `pytest -k pytest -k TestMenuTemplate`

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#166788774](https://www.pivotaltracker.com/story/show/166788774)

### CODE REVIEW

**Comment by RonKbS** [comment]
I've been able to create the template so the feature works. Would recommend a couple of changes to your PR description though;

- You haven't included the need to run `python manage.py create_db` to create the menu_template table. Not sure if it already exists in the environments this will affect, but it did not exist on my local machine.
- I would also recommend changing the README section on how to create tables from `python run.py db migrate` to `python run.py db create_db`
- `"mealPeriod": "lunch"` is missing in the json you're sending.

LGTM otherwise 👍

File: app/repositories/menu_template_repo.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+from app.models.menu_template import MenuTemplate
+from app.repositories import BaseRepo
+
+
+class MenuTemplateRepo(BaseRepo):
+
+    def __init__(self):
+        BaseRepo.__init__(self, MenuTemplate)
```
Comment (tainaandela) [inline app/repositories/menu_template_repo.py]: i think we can call super().__init__ is a python 3 as the recommended approach

File: app/models/menu_template.py [hunk 2]
```diff
@@ -27,16 +28,13 @@ class MenuTemplate(BaseModel):
...
     template_id = db.Column(db.Integer(), db.ForeignKey(
         'menu_template.id'), nullable=False)
 
-    menu_template = db.relationship('MenuTemplate', lazy=False)
-
-    __table_args__ = (
-        db.UniqueConstraint(day, template_id),
-    )
+    menu_template = db.relationship(
+        'MenuTemplate', lazy=False, backref=db.backref('menu_template', lazy='dynamic'))
```
Comment (tainaandela) [inline app/models/menu_template.py]: you are overiding the lazy attribute twice here

File: app/utils/auth.py [hunk 3]
```diff
@@ -25,13 +26,14 @@ def check_token():
         if request.method != 'OPTIONS':
 
             for endpoint in Auth.authentication_header_ignore:
-                if request.path.find(endpoint) > -1: # If endpoint in request.path, ignore this check
+                # If endpoint in request.path, ignore this check
+                if request.path.find(endpoint) > -1:
```
Comment (tainaandela) [inline app/utils/auth.py]: for me just my personal opinion.
```
if endpoint in request.path:
   ...
```
is easier to read

File: app/utils/auth.py [hunk 4]
```diff
@@ -168,16 +173,16 @@ def decorated(*args, **kwargs):
                 user_role = user_role_repo.find_first(**{'user_id': user_id})
 
                 if not user_id:
-                    return make_response(jsonify({'msg': 'Missing User ID in token'})), 400
+                    return make_response(jsonify({'msg': 'Missing User ID in token'})), 401
```
Comment (tainaandela) [inline app/utils/auth.py]: Do you think we should differentiate the proper status code to the front end here.
i.e 400 is unauthorised and 401 is a bad request

Comment (kimbugp) [reply app/utils/auth.py]: @tainaandela, this is addressed in the application's documentation

File: app/models/menu_template.py (hunk 2 above)
Comment (kimbugp) [reply app/models/menu_template.py]: @tainaandela , let me work on this

File: app/repositories/menu_template_repo.py (hunk 1 above)
Comment (kimbugp) [reply app/repositories/menu_template_repo.py]: Noted

**Pull request merged**